### DISCUSSION TITLE
Knockout analysis without specific medium?

### DISCUSSION THREAD

**Discussion by PathogeNish** [micom-dev/micom#130] [Q&A]
Hi All,

I have just started using MICOM's knockout analysis to study the microbial interaction networks in 16S stool samples. By,

- Using the standard `build` function to create a MICOM community
- Using the `com.knockout_taxa` to perform the knockout study

It just occurred to me that nowhere in this have I yet specified the growth medium. Unlike while using the `grow` command to analyse metabolic fluxes where you have to add a set of input fluxes.

My questions are:

1. Doesn't the knockout analysis need a medium to perform the study (the `com.medium` variable looks like it contains the maximum possible flux of every metabolite involved in any reaction for a microbe in the community)?
2. If I force a medium onto the community by changing the `com.medium` variable to the western diet fluxes, and then run the knockout analysis, does this give me more "medium relevant" results?

**Comment by cdiener** [comment]
Hi, sorry for overlooking this. If you don't specify a medium MICOM will use whatever is predefined in the models and that tends to correspond to the richest possible medium (all metabolites with a known import reaction are available). 

I agree that it is probably more relevant to run the knockout analysis with a realistic medium. What you propose in step 2 is exactly how you do it. Just be aware that `com.medium` expects a pandas.Series or dictionary and not a DataFrame like `grow`. So if your medium is a DataFrame you probably want to do something along the lines of `com.medium = medium.flux`.
